velox/exec/HashBuild.cpp

```diff
@@ -319,6 +384,14 @@ void HashBuild::removeInputRowsForAntiJoinFilter() {
 
 void HashBuild::addInput(RowVectorPtr input) {
   checkRunning();
+
+  VELOX_CHECK(!useHashTableCache() || cacheEntry_->builderTaskId == taskId());
+  // For hash table caching: check if table is already cached by another task.
+  // If so, skip processing this input - we'll use the cached table.
+  if (useHashTableCache() && cacheEntry_->buildComplete) {
```

Comment:
I think we shall never run into this? Let's remove or add check that !cacheEntry_->buildComplete